packages/api/src/external/hie/set-doc-query-progress.ts

```diff
@@ -158,7 +159,10 @@ export function aggregateDocProgress(hieDocProgresses: DocumentQueryProgress[]):
   download?: RequiredProgress;
   convert?: RequiredProgress;
 } {
-  const statuses: DocumentQueryStatus[] = [];
+  const statuses: { [key in ProgressType]: DocumentQueryStatus[] } = {
+    download: [],
+    convert: [],
+  };
```

Comment:
Big bug caught thanks to unit testing, we were aggregating the statuses across both download and convert and not separately.